src/R/Components/Impl/ConnectionManager/Implementation/ViewModel/ConnectionViewModel.cs

```diff
@@ -183,15 +187,56 @@ internal sealed class ConnectionViewModel : BindableBase, IConnectionViewModel {
             IsRemote = !(uri?.IsFile ?? true);
         }
 
-       public string GetCompletePath() {
+        public void UpdateName() {
+            var currentPath = Path?.Trim() ?? string.Empty;
+            var previousPath = _previousPath?.Trim() ?? string.Empty;
+
+            var previousProposedName = GetProposedName(previousPath);
+            var currentProposedName = GetProposedName(currentPath);
+
+            // Avoid changing anything if the edit
+            // has no effect on the name (like changing the path's port)
+            if (previousProposedName != currentProposedName) {
+                // Check if the name was calculated from the previous path
+                var currentName = Name ?? string.Empty;
+                if (string.IsNullOrEmpty(currentName) || string.Compare(currentName, previousProposedName, StringComparison.CurrentCultureIgnoreCase) == 0) {
```

Comment:
I can go either way, I have tried both. It felt like it may be more annoying not to track the new name, then to lose a mostly insignificant case change.